### PR TITLE
add option --insecure for oc import-image

### PR DESCRIPTION
**Pull request by wanghaoran1988**
fix https://github.com/openshift/origin/issues/6227

### CODE REVIEW

File: pkg/cmd/cli/cmd/importimage.go [hunk 1]
```diff
@@ -68,7 +69,7 @@ func RunImportImage(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, arg
 
 	from := cmdutil.GetFlagString(cmd, "from")
 	confirm := cmdutil.GetFlagBool(cmd, "confirm")
-
+	insecure := cmdutil.GetFlagBool(cmd, "insecure")
```
Comment (liggitt) [inline pkg/cmd/cli/cmd/importimage.go]: I'd like this to match the annotation better (maybe `--insecure-repository`?) to clarify it is the repository that is insecure, not the API server

Comment (wanghaoran1988) [inline pkg/cmd/cli/cmd/importimage.go]: Good point , will update.

Comment (soltysh) [inline pkg/cmd/cli/cmd/importimage.go]: :+1:

File: pkg/cmd/cli/cmd/importimage.go [hunk 2]
```diff
@@ -98,6 +99,15 @@ func RunImportImage(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, arg
...
 		delete(stream.Annotations, imageapi.DockerImageRepositoryCheckAnnotation)
+		if insecure {
+			stream.Annotations[imageapi.InsecureRepositoryAnnotation] = "true"
+		}
+	} else {
+		if insecure {
+			annotations := make(map[string]string)
+			stream.Annotations = annotations
+			stream.Annotations[imageapi.InsecureRepositoryAnnotation] = "true"
+		}
```
Comment (soltysh) [inline pkg/cmd/cli/cmd/importimage.go]: Instead of repeating the insecure if check do it after the if-else setting annotations, and it'll be only once..

**Comment by wanghaoran1988** [comment]
code updated, please help review again, thanks.

File: pkg/cmd/cli/cmd/importimage.go [hunk 3]
```diff
@@ -45,6 +45,7 @@ func NewCmdImportImage(fullName string, f *clientcmd.Factory, out io.Writer) *co
 	}
 	cmd.Flags().String("from", "", "A Docker image repository to import images from")
 	cmd.Flags().Bool("confirm", false, "If true, allow the image stream import location to be set or changed")
+	cmd.Flags().Bool("insecure-registry", false, "If true, allow the docker registry to be insecure")
```
Comment (soltysh) [inline pkg/cmd/cli/cmd/importimage.go]: @liggitt asked this to reflect the annotation, which is `openshift.io/image.insecureRepository`, thus proposal for `insecure-repository, can you please change this?

File: pkg/cmd/cli/cmd/importimage.go [hunk 4]
```diff
@@ -98,8 +99,16 @@ func RunImportImage(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, arg
 
 	if stream.Annotations != nil {
 		delete(stream.Annotations, imageapi.DockerImageRepositoryCheckAnnotation)
+	} else {
```
Comment (soltysh) [inline pkg/cmd/cli/cmd/importimage.go]: Remove.

File: pkg/cmd/cli/cmd/importimage.go (hunk 3 above)
Comment (wanghaoran1988) [inline pkg/cmd/cli/cmd/importimage.go]: ok, will update

File: pkg/cmd/cli/cmd/importimage.go (hunk 4 above)
Comment (wanghaoran1988) [inline pkg/cmd/cli/cmd/importimage.go]: @soltysh updated

File: pkg/cmd/cli/cmd/importimage.go [hunk 5]
```diff
@@ -98,8 +99,14 @@ func RunImportImage(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, arg
...
+		annotations := make(map[string]string)
```
Comment (soltysh) [inline pkg/cmd/cli/cmd/importimage.go]: One last nit, sorry I didn't spotted that eariler, merge the two lines into one. There's no need for new variable here.

**Comment by soltysh** [comment]
One last nit and you'll be good to go.

**Comment by wanghaoran1988** [comment]
@soltysh updated again . thanks

**Comment by soltysh** [comment]
LGTM [merge]

**Comment by soltysh** [comment]
@wanghaoran1988 thanks for the patience!

**Pull request merged**